pyscriptjs/src/components/pyrepl.ts

```diff
@@ -5,161 +5,201 @@ import { Compartment, StateCommand } from '@codemirror/state';
+            this.appendChild(boxDiv);
+            this.editor.focus();
+            logger.debug(`element ${this.id} successfully connected`);
+        }
 
+        /** Create and configure the codemirror editor
+         */
+        makeEditor(pySrc: string): EditorView {
+            const languageConf = new Compartment();
             const extensions = [
                 indentUnit.of("    "),
                 basicSetup,
                 languageConf.of(python()),
                 keymap.of([
                     ...defaultKeymap,
-                    { key: 'Ctrl-Enter', run: createCmdHandler(this) },
-                    { key: 'Shift-Enter', run: createCmdHandler(this) },
+                    { key: 'Ctrl-Enter',  run: this.execute.bind(this) },
+                    { key: 'Shift-Enter', run: this.execute.bind(this) },
                 ]),
             ];
 
-            if (getEditorTheme(this) === 'dark') {
+            if (getAttribute(this, 'theme') === 'dark') {
                 extensions.push(oneDarkTheme);
             }
 
-            this.editor = new EditorView({
-                doc: this.code.trim(),
+            return new EditorView({
+                doc: pySrc,
                 extensions,
-                parent: this.editorNode,
             });
+        }
 
-            const mainDiv = document.createElement('div');
-            addClasses(mainDiv, ['py-repl-box']);
+        // ******** main entry point for py-repl DOM building **********
+        //
+        // The following functions are written in a top-down, depth-first
+        // order (so that the order of code roughly matches the order of
+        // execution)
+        makeBoxDiv(): HTMLElement {
+            const boxDiv = document.createElement('div');
+            boxDiv.className = 'py-repl-box';
 
-            // Styles that we use to hide the labels whilst also keeping it accessible for screen readers
-            const labelStyle = 'overflow:hidden; display:block; width:1px; height:1px';
+            const editorDiv = this.makeEditorDiv();
+            const editorLabel = this.makeLabel('Python Script Area', editorDiv);
+            this.outDiv = this.makeOutDiv();
 
-            // Code editor Label
-            this.editorNode.id = 'code-editor';
-            const editorLabel = document.createElement('label');
-            editorLabel.innerHTML = 'Python Script Area';
-            editorLabel.setAttribute('style', labelStyle);
-            editorLabel.htmlFor = 'code-editor';
+            boxDiv.append(editorLabel);
+            boxDiv.appendChild(editorDiv);
+            boxDiv.appendChild(this.outDiv);
 
-            mainDiv.append(editorLabel);
+            return boxDiv;
+        }
 
-            // add Editor to main PyScript div
-            mainDiv.appendChild(this.editorNode);
+        makeEditorDiv(): HTMLElement {
+            const editorDiv = document.createElement('div');
+            editorDiv.id = 'code-editor';
```

Comment:
This means no more than one `code-editor` (aka, `py-repl`) per page. I suppose this is intentional... In that case, I'd suggest we don't to that do not force users to have max 1 repl in their apps...

Reply:
I am aware that it's a problem, but it's not a new bug, the main branch has the same logic:
https://github.com/pyscript/pyscript/blob/f9194cc833eb938800f1184574e474bca209dd2d/pyscriptjs/src/components/pyrepl.ts#L81-L86

If anything it means that the new code structure makes existing bugs. But I'm happy to leave the solution to subsequent PRs (possibly not by me 😅).

Reply:
Right.... not sure when this was introduced but, yeah, it's a bug. +1 on fixing it with a follow-up PR